File: csla20vb/Csla/Security/IAuthorizeReadWrite.vb
Namespace Security

  ''' <summary>
  ''' Defines the authorization interface through which an
  ''' object can indicate which properties the current
  ''' user can read and write.
  ''' </summary>
  Public Interface IAuthorizeReadWrite
    ''' <summary>
    ''' Returns <see langword="true" /> if the user is allowed to write the
    ''' to the specified property.
    ''' </summary>
    ''' <returns><see langword="true" /> if write is allowed.</returns>
    ''' <param name="propertyName">Name of the property to read.</param>
    Function CanWriteProperty(ByVal propertyName As String) As Boolean
    ''' <summary>
    ''' Returns <see langword="true" /> if the user is allowed to read the
    ''' specified property.
    ''' </summary>
    ''' <returns><see langword="true" /> if read is allowed.</returns>
    ''' <param name="propertyName">Name of the property to read.</param>
    Function CanReadProperty(ByVal propertyName As String) As Boolean
  End Interface

End Namespace

File: cslacs/Csla/Security/IAuthorizeReadWrite.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Csla.Security
{
  /// <summary>
  /// Defines the authorization interface through which an
  /// object can indicate which properties the current
  /// user can read and write.
  /// </summary>
  public interface IAuthorizeReadWrite
  {
    /// <summary>
    /// Returns <see langword="true" /> if the user is allowed to write the
    /// to the specified property.
    /// </summary>
    /// <returns><see langword="true" /> if write is allowed.</returns>
    /// <param name="propertyName">Name of the property to read.</param>
    bool CanWriteProperty(string propertyName);
    /// <summary>
    /// Returns <see langword="true" /> if the user is allowed to read the
    /// specified property.
    /// </summary>
    /// <returns><see langword="true" /> if read is allowed.</returns>
    /// <param name="propertyName">Name of the property to read.</param>
    bool CanReadProperty(string propertyName);
  }
}
